frontend/gateway/gateway.go

```diff
@@ -224,15 +224,27 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 	return lbf.result, nil
 }
 
-func newLLBBridgeForwarder(ctx context.Context, llbBridge frontend.FrontendLLBBridge, workers frontend.WorkerInfos) (*llbBridgeForwarder, error) {
+func (lbf *llbBridgeForwarder) Result() (*frontend.Result, error) {
```

Comment:
Previously (comment is now collapsed) you said:
> This should probably return a channel that Solve() can wait on (in the case where there is no DeleteBuild()), or be a blocking function.

The channel ended up being in the forwarder (`solver/llbsolver/gateway.gatewayForwarder`) (I confess I'd forgotten about your comment). Should I move it or is it ok as I have it now?

Reply:
> I think the channel would be slightly better in the forwarder. 

I'm confused because it is in the forwarder right now, but perhaps we are talking about different things/places with that name?

Reply:
Ok, yeah everything is called a forwarder :) . I meant it might be better in here as a method of `llbBridgeForwarder`. Maybe we can find a different name for `gatewayForwarder`?

Reply:
"Proxy" is generally pretty overloaded but not (yet? :wink:) in buildkit, so `gatewayForwarder`→`gatewayServerProxy` perhaps? Or since it wraps an `LLBBridgeServer` perhaps `LLBBridgeServerProxy`?

For moving the channel I'll change it to be similar to a `Context` e.g. a `Done() <-chan struct()` and a `Result()`.

Reply:
Is it permissible for frontends to call `Return` more than once? If so are the semantics to keep the first or last answer? 

Currently the code independently keeps the most recent err and result, so you could end up setting both which is probably undesirable...

Reply:
I went with latching the first `Return` but easy enough to change.

Reply:
> Is it permissible for frontends to call Return more than once? 

Not atm but happy to hear use cases. If we keep the restriction we should probably validate it and produce an error if frontend misbehaves.